src/shims/unix/unnamed_socket.rs

```diff
@@ -146,7 +146,7 @@ impl FileDescription for AnonSocket {
             // corresponding ErrorKind variant.
             throw_unsup_format!("reading from the write end of a pipe");
         };
-        let mut readbuf = readbuf.borrow_mut();
+        let readbuf = readbuf.borrow_mut();
         if readbuf.buf.is_empty() {
```

Comment:
```suggestion
        if readbuf.borrow_mut().buf.is_empty() {
```

Then you don't need the explicit drop

Reply:
Nice! I wonder if it would be correct to assume that ``Ref`` or ``RefMut`` will be automatically dropped at the end of the statement if it is not used elsewhere?

Reply:
Yes. Temporary values are dropped at the end of their statement